src/Api/HL/RSQLInput.php

```diff
@@ -0,0 +1,303 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2022 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Api\HL;
+
+use Glpi\Api\HL\Doc as Doc;
+
+class RSQLInput
+{
+    private string $filter_string;
+
+    public function __construct($filter_string)
+    {
+        $this->filter_string = $filter_string;
+    }
+
+    public function getRSQLFilterString(): string
+    {
+        return $this->filter_string;
+    }
+
+    private function rsqlGroupToArray(string $str_array): array
+    {
+        // Convert string like (item1, item2,item3) to array
+        if (!preg_match('/^\((.*)\)$/', $str_array, $matches)) {
+            throw new \InvalidArgumentException('Invalid RSQL group');
+        }
+        $items = explode(',', $matches[1]);
+        // Strip quotes and trim items
+        return array_map(static fn ($item) => trim($item, ' "\''), $items);
```

Comment:
This is dangerous. For instance, if you have a `"teclib'"` item, the ending `'` will be removed. Quotes should be removed only when located on both side, and spaces should be removed only if they are not inside quotes.